### PR TITLE
Upgrade images for jupyterhub-ssh, kbatch

### LINKED ISSUE #1993: CVE vulnerability: update jupyterhub-ssh and kbatch-proxy images
A vulnerability with OpenSSH was recently discovered ([CVE-2023-38408](https://nvd.nist.gov/vuln/detail/CVE-2023-38408)) and the resolution looks like it simply requires that we install a later version of openssh. This is currently affecting the jupyterhub-ssh and kbatch-proxy images.

### PR DESCRIPTION
**Pull request by iameskild**
## Reference Issues or PRs

Closes #1993 

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [x] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [ ] Did you add new tests?

### CODE REVIEW

File: src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub-ssh/ssh.tf [hunk 1]
```diff
@@ -11,7 +11,14 @@ resource "kubernetes_config_map" "jupyterhub-ssh" {
   }
 
   data = {
-    "values.yaml" = "hubUrl: ${var.jupyterhub_api_url}"
+    "values.yaml" = <<-EOT
+      hubUrl: ${var.jupyterhub_api_url}
+      ssh:
+        config:
+          JupyterHubSSH:
+            debug: true
```
Comment (iameskild) [inline src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub-ssh/ssh.tf:19]: As workaround for https://github.com/yuvipanda/jupyterhub-ssh/issues/80

File: src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub-ssh/ssh.tf [hunk 2]
```diff
@@ -11,7 +11,14 @@ resource "kubernetes_config_map" "jupyterhub-ssh" {
...
+            host_key_path: /etc/jupyterhub-ssh/secrets/jupyterhub-ssh.host-key
```
Comment (iameskild) [inline src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub-ssh/ssh.tf:20]: I also kept running into this issue and specifying the `host_key_path` explicitly was the only way around this:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/srv/jupyterhub-ssh/jupyterhub_ssh/__main__.py", line 4, in <module>
    main()
  File "/srv/jupyterhub-ssh/jupyterhub_ssh/__init__.py", line 331, in main
    loop.run_until_complete(app.start_server())
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/srv/jupyterhub-ssh/jupyterhub_ssh/__init__.py", line 315, in start_server
    await asyncssh.listen(
  File "/home/jovyan/.local/lib/python3.8/site-packages/asyncssh/connection.py", line 8310, in listen
    new_options = cast(SSHServerConnectionOptions, await _run_in_executor(
  File "/home/jovyan/.local/lib/python3.8/site-packages/asyncssh/connection.py", line 515, in _run_in_executor
    return await loop.run_in_executor(
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/jovyan/.local/lib/python3.8/site-packages/asyncssh/connection.py", line 6463, in __init__
    super().__init__(options=options, last_config=last_config, **kwargs)
  File "/home/jovyan/.local/lib/python3.8/site-packages/asyncssh/misc.py", line 350, in __init__
    self.prepare(**self.kwargs)
  File "/home/jovyan/.local/lib/python3.8/site-packages/asyncssh/connection.py", line 7803, in prepare
    server_keys = load_keypairs(server_host_keys, passphrase,
  File "/home/jovyan/.local/lib/python3.8/site-packages/asyncssh/public_key.py", line 3476, in load_keypairs
    read_private_key_and_certs(key_to_load, passphrase)
  File "/home/jovyan/.local/lib/python3.8/site-packages/asyncssh/public_key.py", line 3290, in read_private_key_and_certs
    key, cert = import_private_key_and_certs(read_file(filename), passphrase)
  File "/home/jovyan/.local/lib/python3.8/site-packages/asyncssh/misc.py", line 221, in read_file
    with open_file(filename, mode) as f:
  File "/home/jovyan/.local/lib/python3.8/site-packages/asyncssh/misc.py", line 203, in open_file
    return open(Path(filename).expanduser(), mode, buffering=buffering)
IsADirectoryError: [Errno 21] Is a directory: '.'
Stream closed EOF for dev/nebari-jupyterhub-ssh-78b9796d6c-xdzgt (jupyterhub-ssh)
```

File: src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub-ssh/sftp.tf [hunk 3]
```diff
@@ -90,7 +90,8 @@ resource "kubernetes_deployment" "jupyterhub-sftp" {
         volume {
           name = "secrets"
           secret {
-            secret_name = kubernetes_secret.jupyterhub-sftp.metadata.0.name
+            secret_name  = kubernetes_secret.jupyterhub-sftp.metadata.0.name
+            default_mode = "0600"
```
Comment (iameskild) [inline src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub-ssh/sftp.tf:94]: Something changed and I'm not sure where but I needed to explicitly restrict the permissions on this file due to the following error:

```
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@         WARNING: UNPROTECTED PRIVATE KEY FILE!          @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
Permissions 0644 for '/etc/jupyterhub-sftp/config/hostKey' are too open.
It is required that your private key files are NOT accessible by others.
This private key will be ignored.
Unable to load host key "/etc/jupyterhub-sftp/config/hostKey": bad permissions
Unable to load host key: /etc/jupyterhub-sftp/config/hostKey
sshd: no hostkeys available -- exiting.
Stream closed EOF for dev/nebari-jupyterhub-sftp-5c7b95d76f-gmdps (jupyterhub-sftp)
```

**Comment by iameskild** [comment]
Hmm jupyterhub-sftp is still running an older version of OpenSSH. I will need to dig into why the latest version wasn't installed...

File: src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub-ssh/sftp.tf (hunk 3 above)
Comment (Adam-D-Lewis) [reply src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub-ssh/sftp.tf:94]: Maybe the new version of OpenSSH refuses to run if the file has the wrong permissions now where the old one only raised a warning

File: src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub-ssh/ssh.tf (hunk 1 above)
Comment (Adam-D-Lewis) [reply src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub-ssh/ssh.tf:19]: It seems strange to set debug to true, but I checked and that is the default value - https://github.com/yuvipanda/jupyterhub-ssh/blob/34cbd1813738299b08705231080d84dcaee56aad/helm-chart/jupyterhub-ssh/values.yaml#L31

**Review by Adam-D-Lewis — APPROVED** [review]

**Comment by Adam-D-Lewis** [comment]
LGTM!

File: src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub-ssh/ssh.tf (hunk 1 above)
Comment (iameskild) [reply src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub-ssh/ssh.tf:19]: The `debug` isn't strictly necessary but including the `ssh.config` in the values.yaml is due to the issue I linked to above.

**Pull request merged**